### DISCUSSION TITLE
Requirements to host mitm.html

### DISCUSSION THREAD

**Discussion by john-v-k** [jimmywarting/StreamSaver.js#230] [General]
I'm a bit new to streamsaver so, sorry if this is a dumb question. What is mitm.html file? And if I want to host my own mitm.html, how do I do that?

**Comment by jimmywarting** [comment]
The mitm is not for spying on what you download - all doe i could just change it at any given point and sneak something in there, but that would never happen. I would be violating so many rules. I don't even have any Server side logs or google analytics that tracks IP, referer, online time etc

Mitm only lazy loaded in a hidden iframe once the download begins.

It's a mean of transferring a MessageChannel from a the main thread to a external https site that installs a service worker and mimic what a server dose to download stuff without intervene in your own local service worker (if you have that)
It was also necessary to have it on insecure sites and jsfiddle's that could not install service workers. It made it easier for ppl to just require streamsaver from npm and start using it without any additional setups

If service worker wasn't required then there would be any need for for a sw.js or a mitm.html
if insecure sites could also install service worker, then i maybe would have created the mitm in the first place, maybe i would have written a guide to how to install/import the necessary code for how to put up a service worker and communicated to the service worker directly with postMessages instad of via a "proxy"

To change it you to your own you would also need to host the mitm.html:

```js
// StreamSaver can detect and use the Ponyfill that is loaded from the cdn.
streamSaver.WritableStream = streamSaver.WritableStream
streamSaver.TransformStream = streamSaver.TransformStream
// if you decide to host mitm + sw yourself
streamSaver.mitm = 'https://example.com/custom_mitm.html'
```

**Reply by john-v-k** [reply]
Thanks for the response. The reason I was asking is because I wanted it to work even after disabling third party cookies. Is there any way I could do it?

**Reply by john-v-k** [reply]
I want to host my own mitm.html and sw.js. Could you please give an example of how to do that?

**Reply by jimmywarting** [reply]
if you don't want to use the mitm, and host everything yourself - then i recommend to try out https://www.npmjs.com/package/native-file-system-adapter

**Reply by john-v-k** [reply]
No, I want to use it but when i did `php -S localhost:3001` and put mitm.html and sw.js in a folder and change mitm, it is not working.

**Reply by jimmywarting** [reply]
can you share some code where you use streamsaver.js?
what isn't working, do you have any stack trace?

if you are on discord then we can chat more quickly 
Endless#1510

**Reply by john-v-k** [reply]
Sorry for the delayed response. GitHub isn't sending notifications for discussions so that's why. Anyway tnx for the quick response and I got it working :) I made a very silly error wrt file paths. Thanks!
